### PR TITLE
Add cairo1 prank cheatcode

### LINKED ISSUE #1748: Add `prank` cheatcode
It also should include stop variant 

```
let x = prank(target_address, value)
stop_prank(x)
```

### PR DESCRIPTION
**Pull request by cptartur**
Closes #1748

Docs https://github.com/software-mansion/protostar/pull/1811

### CODE REVIEW

**Comment by cptartur** [comment]
Docs here https://github.com/software-mansion/protostar/pull/1811

File: tests/integration/cairo1_hint_locals/prank/prank_test.cairo [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    assert(*return_data.at(0_u32) == 2, 'check call result');
+}
+
+#[test]
+fn test_stop_prank_multiple_times() {
+    let deployed_contract_address = deploy_contract('pranked', ArrayTrait::new()).unwrap();
+    start_prank(123, deployed_contract_address).unwrap();
+
+    stop_prank(deployed_contract_address).unwrap();
+    stop_prank(deployed_contract_address).unwrap();
```
Comment (Arcticae) [inline tests/integration/cairo1_hint_locals/prank/prank_test.cairo:50]: shouldn't that fail?

File: tests/integration/cairo1_hint_locals/prank/prank_test.cairo [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    assert(*return_data.at(0_u32) == 2, 'check call result');
+}
+
+#[test]
+fn test_stop_prank_on_non_existent() {
+    stop_prank(1234).unwrap();
+}
+
+#[test]
+fn test_stop_prank_on_not_pranked() {
```
Comment (Arcticae) [inline tests/integration/cairo1_hint_locals/prank/prank_test.cairo:34]: Those two should fail i believe (be broken)

File: tests/integration/cairo1_hint_locals/prank/prank_test.cairo [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    let return_data = call(deployed_contract_address, 'get_a', ArrayTrait::new()).unwrap();
+    assert(*return_data.at(0_u32) == 3, 'check call result');
+}
+
+#[test]
+fn test_stop_prank_cancels_all_pranks() {
+    let deployed_contract_address = deploy_contract('pranked', ArrayTrait::new()).unwrap();
+
+    start_prank(123, deployed_contract_address).unwrap();
+    start_prank(123, deployed_contract_address).unwrap();
```
Comment (Arcticae) [inline tests/integration/cairo1_hint_locals/prank/prank_test.cairo:77]: Pranking twice shouldn't really be different to pranking once, we store only a key in a map, so this test doesn't make much sense

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: tests/integration/cairo1_hint_locals/prank/prank_test.cairo (hunk 3 above)
Comment (cptartur) [reply tests/integration/cairo1_hint_locals/prank/prank_test.cairo:77]: I think it useful to have test like that. There should be nothing preventing the user from pranking the same address multiple times.

File: tests/integration/cairo1_hint_locals/prank/prank_test.cairo (hunk 1 above)
Comment (cptartur) [reply tests/integration/cairo1_hint_locals/prank/prank_test.cairo:50]: Previously `start_prank` returned a callable that could cancel a prank. I'm assuming it was impossible for it to fail then. 

Having the ability to call `stop_prank` multiple times/on non-existing is just the behavior I chose.

Comment (Arcticae) [reply tests/integration/cairo1_hint_locals/prank/prank_test.cairo:50]: If others agree about this, i'm fine with it as well, don't have a strong opinion on this

File: protostar/cheatable_starknet/callable_hint_locals/start_prank_hint_local.py [hunk 4]
```diff
@@ -13,13 +13,13 @@ def __init__(self, contracts_controller: ContractsController):
...
     def name(self) -> str:
-        return "prank"
+        return "start_prank"
 
     def _build(self) -> Callable:
         return self.prank
 
-    def prank(self, caller_address: int, target_address: int):
-        valid_target_address = Address.from_user_input(target_address)
+    def prank(self, caller_address: int, target_contract_address: int):
```
Comment (MaksymilianDemitraszek) [inline protostar/cheatable_starknet/callable_hint_locals/start_prank_hint_local.py:21]: so `prank` or `start_prank`, please commit and use one consistently 🥺

File: tests/integration/cairo1_hint_locals/prank/prank_test.cairo [hunk 5]
```diff
@@ -0,0 +1,88 @@
+use array::ArrayTrait;
```
Comment (MaksymilianDemitraszek) [inline tests/integration/cairo1_hint_locals/prank/prank_test.cairo:1]: ❤️ Very solid suite

File: tests/integration/cairo1_hint_locals/prank/prank_test.cairo (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply tests/integration/cairo1_hint_locals/prank/prank_test.cairo:50]: In my opinion 
- pranking non existent contract makes sense, you might want to prank constructor on deploy
- stop_prank - when it cleans up everything i would also let it run on unpranked contracts without error. I interpret the semantics as - clean everything up if there is something to clean up

**Review by MaksymilianDemitraszek — DISMISSED** [review]
Please do not split the docs and code in future PRs

**Review by karol-bisztyga — DISMISSED** [review]
this is alright 👍

**Review by Arcticae — DISMISSED** [review]

File: protostar/cheatable_starknet/callable_hint_locals/start_prank_hint_local.py (hunk 4 above)
Comment (Arcticae) [reply protostar/cheatable_starknet/callable_hint_locals/start_prank_hint_local.py:21]: `start_prank` vs `prank` semantics are from foundry cheatcodes, if we use the names we should stick to what they do in foundry as well.

File: tests/integration/cairo1_hint_locals/prank/prank_test.cairo (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply tests/integration/cairo1_hint_locals/prank/prank_test.cairo:50]: Good point @Arcticae, but I don't see a reason why to not add it here

File: protostar/cheatable_starknet/callable_hint_locals/start_prank_hint_local.py (hunk 4 above)
Comment (cptartur) [reply protostar/cheatable_starknet/callable_hint_locals/start_prank_hint_local.py:21]: It is leftover from the previous code. I will change it to `start_prank` here.

File: tests/integration/cairo1_hint_locals/prank/prank_test.cairo (hunk 2 above)
Comment (cptartur) [reply tests/integration/cairo1_hint_locals/prank/prank_test.cairo:34]: We agreed to allow stoping prank on non-pranked/non-existent addreses.

**Comment by cptartur** [comment]
I merged the docs into this PR for easier review.

**Review by MaksymilianDemitraszek — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

**Review by karol-bisztyga — APPROVED** [review]

**Pull request merged**